sips/instance_decided_enforcement.md

```diff
@@ -0,0 +1,134 @@
+| Author      | Title                        | Category | Status |
+|-------------|------------------------------|----------|--------|
+| Alon Muroch | Instance Decided Enforcement | Core     | open-for-discussion  |
+
+**Summary** 
+Describes the remove of requiring a QBFT instance to decided to start the next one. 
+
+**Rational**  
+Currently QBFT instances have an incremental index for their height, starting with 0.
+For a new instance to start, the QBFT controller checks that the previous one has decided.
+Originally this was made so instance heights are predictable, synchronized and easy to sync by every node in the network.
+
+The main reasons to drop these requirements are:
+1) Instance Height can be the duty's slot number as every duty, at most, there is 1 duty per type of duty per validator.
+2) Syncing historical decided instances can be made easy with the use of epochs, e.g. fetch decided instances for epochs 100-150.
+3) Stalled/ non decided duties have a life expectancy of, at most, 2 epochs, since past that even if executed will not gain rewards for validator (per beacon roles)
+4) There is a difficult balance between efficient constant time timeouts and liveness. There are some complex solutions but since duties have an effective expiration time and are "non-dependent" we can terminate at a certain round.
+
+**Spec Changes** 
+
+**QBFT Controller**
+- Remove function CanStartInstance
+- Remove function canStartInstanceForValue
+- Remove function bumpHeight
+- Remove future message processing - no need as each instance is independent. If a future decided message is received it will trigger decided sync
+```go
+// StartNewInstance will start a new QBFT instance, if can't will return error
+func (c *Controller) StartNewInstance(height Height, value []byte) error {
+	if err := c.GetConfig().GetValueCheckF()(value); err != nil {
+		return errors.Wrap(err, "value invalid")
+	}
+
+	// only if current height's instance exists (and decided since passed can start instance) bump
+	if c.StoredInstances.FindInstance(height) != nil {
+		return errors.New("instance already running")
+	}
+
+	newInstance := c.addAndStoreNewInstance()
+	newInstance.Start(value, height)
+
+	c.Height = height
+
+    c.forceStopAllInstanceExceptCurrent()
+
+	return nil
+}
```

Comment:
Maybe we should have a check that the new instance's height is larger than any stored instance.
```
if height <= c.Height {
     return errors.New("past height")
```

Imagine a bug in a syncing beacon node that for some weird reason gives the nodes duties out of order.

Reply:
good catch, fixed